### PR TITLE
[Bug fix] Main branch issue

### PR DESCRIPTION
**Pull request by mugx-fc**
This PR tries to fix the issue that comes when the main branch is called `main` and not `master`.

### CODE REVIEW

**Comment by fc-dropzone-github** [comment]
### :lock: Prisma Cloud scan details - vulnerabilities :warning:
---
<details open>
<summary><b>:white_check_mark: Passed <code>shipment_tracker:latest</code></b><br />&emsp;&emsp;&ensp;Vulnerabilities found. Total: 0, critical: 0, high: 0, medium: 0, low: 0.</summary><br />




#### No vulnerabilities found :tada:

</details>

File: app/models/payloads/github_pull_request.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def branch_name
     end
 
     def base_branch_master?
-      pull_request&.dig('base', 'ref') == 'master'
+      pull_request&.dig('base', 'ref') == 'master' || pull_request&.dig('base', 'ref') == 'main'
```
Comment (LS80) [inline app/models/payloads/github_pull_request.rb:62]:
```suggestion
      %w(main master).include? pull_request&.dig('base', 'ref')
```
That's probably more idiomatic Ruby?

Comment (mugx-fc) [reply app/models/payloads/github_pull_request.rb]: yep, even better, thanks 👍

**Review by LS80 — APPROVED** [review]

**Review by 0mega — COMMENTED** [review]
Hey @mugx-fc, nice to see some love for this repo.

I can see that this is the second attempt to bring `main` branch support and it would be great to get it right. These things are on my mind:

- There are several levels of tests which specify the behaviour of this project regarding branching. We should have an update to those to reflect new functionality. That's the easiest way one has to actually prove that the fix works.
- There are more than a 100 references to `master` in the codebase which would be nice to rename, not a must functionality-wise but if we are serious about fixing this then we should at least have a follow-up PR/Issue for renaming it all.
- There are rake tasks that produce CSV reports based on `master` branch which should also be updated
- Supporting `main` branch actually smells like a subset of the larger problem of supporting custom default branches on GitHub, so instead of hardcoding another value, a different approach could be to provide an array of branches as configuration to this project and provide a friendlier usability.

Let me know your thoughts @mugx-fc, @LS80, @bliof.

**Comment by mugx-fc** [comment]
@0mega agreed on all your points 👍

**Pull request merged**